### PR TITLE
* Basic support for T-SQL query hints.

### PR DESCRIPTION
**Pull request by SLeopold1986**
With this patch a basic support for T-SQL query hints is provided. It is possible to use:

`.WithOptions(u => u.SetHint("RECOMPILE"))`

### CODE REVIEW

File: src/NHibernate/Linq/IQueryableOptions.cs [hunk 1]
```diff
@@ -38,5 +38,12 @@ public interface IQueryableOptions
 		/// <param name="timeout">The timeout in seconds.</param>
 		/// <returns><see langword="this" /> (for method chaining).</returns>
 		IQueryableOptions SetTimeout(int timeout);
+
+		/// <summary>
+		/// Set a T-SQL Query hint as an Option to the query
+		/// </summary>
+		/// <param name="hint">The t-sql query hint</param>
+		/// <returns><see langword="this" /> (for method chaining).</returns>
+		IQueryableOptions SetHint(string hint);
```
Comment (hazzik) [inline src/NHibernate/Linq/IQueryableOptions.cs:47]: This interface should not be changed. It is obsolete.

File: src/NHibernate/Loader/Hql/QueryLoader.cs [hunk 2]
```diff
@@ -104,6 +104,14 @@ public override bool IsSubselectLoadingEnabled
 			return dialect.ApplyLocksToSql(sql, aliasedLockModes, keyColumnNames);
 		}
 
+		protected override SqlString ApplyHint(SqlString sql, string hint, Dialect.Dialect dialect)
```
Comment (hazzik) [inline src/NHibernate/Loader/Hql/QueryLoader.cs:107]: This method is not used anywhere.

File: src/NHibernate/Loader/Loader.cs [hunk 3]
```diff
@@ -227,6 +227,16 @@ protected virtual SqlString ApplyLocks(SqlString sql, IDictionary<string, LockMo
 			return sql;
 		}
 
+		/// <summary>
+		/// Append <c> OPTION(...hints...)</c> clause, if necessary. This
+		/// empty superclass implementation merely returns its first
+		/// argument.
+		/// </summary>
+		protected virtual SqlString ApplyHint(SqlString sql, string hints, Dialect.Dialect dialect)
```
Comment (hazzik) [inline src/NHibernate/Loader/Loader.cs:235]: This method is not used anywhere.

File: src/NHibernate/Loader/Hql/QueryLoader.cs [hunk 4]
```diff
@@ -343,9 +351,9 @@ protected override IResultTransformer ResolveResultTransformer(IResultTransforme
 			Object[] resultRow = GetResultRow(row, rs, session);
 			bool hasTransform = HasSelectNew || resultTransformer != null;
 			return (!hasTransform && resultRow.Length == 1
-				        ? resultRow[0]
```
Comment (hazzik) [inline src/NHibernate/Loader/Hql/QueryLoader.cs:346]: Please revert these changes.

File: src/NHibernate/Loader/Hql/QueryLoader.cs [hunk 5]
```diff
@@ -441,7 +449,7 @@ internal IEnumerable GetEnumerable(QueryParameters queryParameters, IEventSource
 			var rs = GetResultSet(cmd, queryParameters, session, null);
 
 			var resultTransformer = _selectNewTransformer ?? queryParameters.ResultTransformer;
-			IEnumerable result = 
+			IEnumerable result =
```
Comment (hazzik) [inline src/NHibernate/Loader/Hql/QueryLoader.cs:452]: Please revert this changes

File: src/NHibernate/Engine/RowSelection.cs [hunk 6]
```diff
@@ -62,5 +63,11 @@ public bool DefinesLimits
 		{
 			get { return maxRows != NoValue || firstRow > 0; }
 		}
+
+		public string Hint
```
Comment (hazzik) [inline src/NHibernate/Engine/RowSelection.cs:67]: This is a strange place to add query hint...

**Comment by hazzik** [comment]
The PR is incomplete. The hints are set, but not used by anything. And it lacks tests. We cannot accept PRs without tests

File: src/NHibernate/Engine/RowSelection.cs (hunk 6 above)
Comment (SLeopold1986) [reply src/NHibernate/Engine/RowSelection.cs:67]: What would be a better place?

**Comment by hazzik** [comment]
I'm going to close this PR as it is incomplete as far as I concern.
